src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/Api/ServiceController.php

```diff
@@ -0,0 +1,117 @@
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Surfnet\ServiceProviderDashboard\Application\Assembler\ServiceStatusAssembler;
+use Surfnet\ServiceProviderDashboard\Application\Command\Service\CreateServiceCommand;
+use Surfnet\ServiceProviderDashboard\Application\Command\Service\EditServiceCommand;
+use Surfnet\ServiceProviderDashboard\Application\Exception\EntityNotFoundException;
+use Surfnet\ServiceProviderDashboard\Application\Exception\InvalidArgumentException;
+use Surfnet\ServiceProviderDashboard\Application\Service\ServiceService;
+use Surfnet\ServiceProviderDashboard\Application\Service\ServiceStatusService;
+use Surfnet\ServiceProviderDashboard\Infrastructure\DashboardBundle\Command\Service\SelectServiceCommand;
+use Surfnet\ServiceProviderDashboard\Infrastructure\DashboardBundle\Form\CreateServiceType;
+use Surfnet\ServiceProviderDashboard\Infrastructure\DashboardBundle\Form\EditServiceType;
+use Surfnet\ServiceProviderDashboard\Infrastructure\DashboardBundle\Service\AuthorizationService;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class ServiceController extends Controller
+{
+    /**
+     * @var CommandBus
+     */
+    private $commandBus;
+
+    /**
+     * @var AuthorizationService
+     */
+    private $authorizationService;
+
+    /**
+     * @var ServiceService
+     */
+    private $serviceService;
+
+    /**
+     * @var ServiceStatusService
+     */
+    private $serviceStatusService;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @param RouterInterface $router
+     * @param CommandBus $commandBus
+     * @param AuthorizationService $authorizationService
+     * @param ServiceService $serviceService
+     * @param ServiceStatusService $serviceStatusService
+     */
+    public function __construct(
+        RouterInterface $router,
+        CommandBus $commandBus,
+        AuthorizationService $authorizationService,
+        ServiceService $serviceService,
+        ServiceStatusService $serviceStatusService
+    ) {
+        $this->router = $router;
+        $this->commandBus = $commandBus;
+        $this->authorizationService = $authorizationService;
+        $this->serviceService = $serviceService;
+        $this->serviceStatusService = $serviceStatusService;
+    }
+
+    /**
+     * @Method({"GET", "POST"})
+     * @Route("/api/service/status", name="api_service_status")
+     * @Security("has_role('ROLE_ADMINISTRATOR')")
+     * @Template()
+     *
+     * @param Request $request
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
+     */
+    public function statusAction(Request $request)
+    {
+        $service = $this->serviceService->getServiceById(
```

Comment:
The idea is that a user (known as a Contact) can have 1..n Services. The services of the user can be retrieved from the authorizationservice::getAllowedServiceNamesById.

Do we want to make this a singular get endpoint where you pass an identifier of the service?
Or do we want to search all available services for the user and return them all at once?

My personal preference would be the last option.

Reply:
I think it would not be an issue to determine this when building the 'donut'. If it's changed this would be a very small change.